src/goto-analyzer/show_on_source.cpp

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************\
+
+Module: goto-analyzer
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+#include "show_on_source.h"
+
+#include <util/file_util.h>
+#include <util/message.h>
+#include <util/unicode.h>
+
+#include <analyses/ai.h>
+
+#include <fstream>
+
+/// get the name of the file referred to at a location loc,
+/// if any
+static optionalt<std::string>
+show_location(const ai_baset &ai, ai_baset::locationt loc)
+{
+  const auto abstract_state = ai.abstract_state_before(loc);
+  if(abstract_state->is_top())
+    return {};
```

Comment:
This doesn't match the name of the function or the documentation; why should the file name depend on the domain.